[Soh Jun Qi] iP

sjq-sohjunqi opened the pull request.

sjq-sohjunqi committed:
Level 1

sjq-sohjunqi committed:
Update list storage to Chatbot

Chatbot does not remember user input

A list will provide the chatbot with more functionality

Let's store user input in a list for future reference

sjq-sohjunqi committed:
Convert to task list

List only stored string names without meaning

Adding ability to mark objects as done allows them to be treated as tasks.

sjq-sohjunqi committed:
Revert "Convert to task list"

This reverts commit 4eb206371ef455fc5a735f772e03fe214feb7887.

sjq-sohjunqi committed:
Convert to task list

List only stored string names without meaning

Adding ability to mark objects as done allows them to be treated as tasks.

sjq-sohjunqi committed:
Convert to task list

List only stored string names without meaning

Adding ability to mark objects as done allows them to be treated as tasks.

sjq-sohjunqi committed:
Add task types

Different types of tasks not differentiated

By introducing task types, different tasks can have specialised characteristics relevant to them

sjq-sohjunqi committed:
Automate Text UI Testing

Update scripts for automated text UI testing

sjq-sohjunqi committed:
Add exception handling

Exceptions were not handled correctly

Added exception handling with specific error messages to aid user in identifying mistakes

sjq-sohjunqi committed:
Add delete function in task list

User is unable to remove old tasks from the list

Let's add a delete function to the task list

sjq-sohjunqi committed:
Add Save feature

Duke does not save the tasklist of users so each session is independent of another

Storing task information into a file allows Duke to retain information from previous sessions

sjq-sohjunqi committed:
Add date recognition

Duke only treats dates as strings and will not validate user input

Duke now only accepts properly formatted date inputs

sjq-sohjunqi committed:
Implement more OOP

Most of Duke's logic was coded into the chatbox function.

Seperated logic into different components to make full use of OOP in Java

sjq-sohjunqi committed:
Moved into package

sjq-sohjunqi committed:
Add JUnit tests

Manually entered test cases and verified with provided

Now can partially automate testing with help of JUnit

sjq-sohjunqi committed:
Add JavaDoc comments

JavaDoc header comments added to give specificity to the purpose of each function

sjq-sohjunqi committed:
Revert "Add JavaDoc comments"

This reverts commit e9b8ee35d0f5a50221e624340cafbb4261dfa414.

sjq-sohjunqi committed:
Add JavaDoc comments

JavaDoc header comments added to give specificity to the purpose of each function

sjq-sohjunqi committed:
Edit code to follow convention

Edit code to follow convention

The code did not follow the coding conventions, making it harder to read.

Let's change code to follow standard Java coding standards.

sjq-sohjunqi committed:
Add find feature

Add find feature

User can only display all items of the list. This can make it hard to look for a specific item when the list is long.

Let's add a find feature which will search the list for the items containing the search term.

sjq-sohjunqi committed:
Revert "Add find feature"

This reverts commit 7dd94050efe31c14cf79c926945c4778667c2f93.

sjq-sohjunqi committed:
Add find feature

Add find feature

User can only display all items of the list. This can make it hard to look for a specific item when the list is long.

Let's add a find feature which will search the list for the items containing the search term.

sjq-sohjunqi committed:
Merge branch 'branch-A-JavaDoc'

sjq-sohjunqi committed:
Merge branch 'branch-A-CodingStandard'

sjq-sohjunqi committed:
Resolved branch merge conflicts

sjq-sohjunqi committed:
Merge branch 'branch-Level-9'

sjq-sohjunqi committed:
Add Gradle Support

Add Gradle Support

Graddle will help to automate some of the build tasks of the project

Let's add support of Gradle to the project

sjq-sohjunqi committed:
Add GUI

Add GUI

Duke was using text-based UI to interact with user.

Let's add JavaFX to make the UI more user-friendly

glatiuden commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+    * @TaskList the TaskList object the commands will be processed on
+    * @return String object containing results of processing 
+    */
+    public String process(String in, TaskList tl){
```
Maybe the method name can be changed slightly to explain the method more intuitively? E.g. processInputToAction? There should be a space between `)` and `{`. I noticed this on your other methods and some of the if statements as well.

glatiuden commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+
+            String res = "";
+            
+            String[] split = in.split("\\s");
```
Perhaps a more meaningful name can be given to this variable in order to avoid confusion, and preferably in plural form since it is a String collection? E.g. splitStrings

glatiuden commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+    private boolean end;
```
Would it be better if it was `isEnd`? For making a boolean variable "sounds like" boolean. 
```suggestion
    private boolean isEnd;
```

glatiuden commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,128 @@
+        
+        String str = "["+type+"]";
+        
+        if (done)
+            str += "[X] "+name;
+        else
+            str += "[ ] "+name;
```
Java coding standard stated that: `Single statement conditionals should still be wrapped by curly brackets`. Perhaps you could tweak it a little bit? I noticed the same issue on some of the other if-else statements as well.

glatiuden reviewed (commented):
LGMT. Just a few comments on the code quality guidelines.

Cp-John commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,20 @@
+        displayPicture = iv;
+
+        text.setWrapText(true);
+
```
Are the blank lines here necessary?

Cp-John commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,190 @@
+     */
+    @Override
+    public void start(Stage stage) {
+        Parser p = new Parser();
```
Why don't you use the full name parser as the variable name? The same for others.

Cp-John commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+                LocalDate dt = LocalDate.now();
+                
+                try {
+                
```
Why do you put a blank line here?

Cp-John commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+    */
+    public String process(String in, TaskList tl){
+        
+        try {
```
Each case block is too long here. Perhaps abstract part of them into methods with reasonable names which increases readability of your code.

Cp-John commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,82 @@
+package duke;
+
+import duke.*;
```
Remember to always import classes explicitly according to Java coding standard.

Cp-John reviewed (commented):
LGTM. But there are several unnecessary blank lines.

sjq-sohjunqi committed:
Fix Coding Style Issues

Fix Coding Style Issues

As per several pull request comments on Github, Duke had several coding style issues that could be ironed out to improve readability.

Let's fix those coding issues.

sjq-sohjunqi committed:
Add assertions

Add assertions

Certain areas of code might fail without the program realising. Specifically, it is not checked whether the addition and removal of tasks were executed without fail.

Let's add assertions to the addition of tasks and removal of tasks.

sjq-sohjunqi committed:
Improve code quality

Improve code quality

There were still some parts of code inside TaskList class that did not adhere to coding standards.

Let's update that to fit coding standards.

sjq-sohjunqi committed:
Merge pull request #1 from sjq-sohjunqi/branch-A-Assertions

Add assertions

sjq-sohjunqi committed:
Merge branch 'master' into branch-A-CodeQuality

sjq-sohjunqi committed:
Merge pull request #2 from sjq-sohjunqi/branch-A-CodeQuality

Improve code quality

sjq-sohjunqi committed:
Add reminders

Add reminders

Users were unable to be informed when a deadline or event is nearing its due date.

Let's add a feature to remind users at the start of the program.

sjq-sohjunqi committed:
Add better GUI

Text from the program and the user is not split and can be confusing.

Let's move all user input to one side of the screen and the program's output to the other.

sjq-sohjunqi committed:
Add image of UI

Let's add an image of user interface

sjq-sohjunqi committed:
Set theme jekyll-theme-cayman

sjq-sohjunqi committed:
Fix bug with pre-exisiting tasklist

Program does not check before attempting to get check the dates associated with the task. It will encounter an error when a task without date is checked.

Let's add a check to skip over tasks without dates.

sjq-sohjunqi committed:
Add user guide

Let's add a user guide to tell the user how to use Duke.

sjq-sohjunqi committed:
Update User Guide

Let's add new line to each expected outcome

sjq-sohjunqi committed:
Update User Guide

Let's add new line to each expected outcome

sjq-sohjunqi committed:
Update User Guide

Let's add new lines to each expected outcome

sjq-sohjunqi committed:
Update User Guide

Let's add new lines to each expected outcome

sjq-sohjunqi committed:
Update User Guide

Let's add new lines to each expected outcome
